src/openms/source/ANALYSIS/TOPDOWN/DeconvolutedSpectrum.cpp

```diff
@@ -0,0 +1,456 @@
+           << "PRECURSOR_INTENSITY=" << precursorPeak.getIntensity() << "\n";
+      }
+      else
+      {
+        auto avgMass = (precursorPeak.getMZ() - FLASHDeconvHelperStructs::getChargeMass(precursorPeak.getCharge()>0)) * abs(precursorPeak.getCharge());
+        auto mMass = avgMass - avg.getAverageMassDelta(avgMass);
+        fs << "MS_ONE_ID=" << 0 << "\n"
+            << "MS_ONE_SCAN=" << precursorScanNumber << "\n"
+            << "PRECURSOR_MZ="
+            << std::to_string(precursorPeak.getMZ()) << "\n"
+            << "PRECURSOR_CHARGE=" << precursorPeak.getCharge() << "\n"
+            << "PRECURSOR_MASS=" << std::to_string(mMass) << "\n"
+            << "PRECURSOR_INTENSITY=" << precursorPeak.getIntensity() << "\n";
+      }
+    }
+    fs << std::setprecision(-1);
+
+    double scoreThreshold = 0;
+    std::vector<double> scores;
+
+    if (size() > 500)// max peak count for TopPic
+    {
+      scores.reserve(size());
+      for (auto &pg : *this)
+      {
+        scores.push_back(pg.getIsotopeCosine());
+      }
+      std::sort(scores.begin(), scores.end());
+      scoreThreshold = scores[scores.size() - 500];
+      std::vector<double>().swap(scores);
+    }
+
+    int size = 0;
+    for (auto &pg : *this)
+    {
+      if (pg.getIsotopeCosine() < scoreThreshold)
+      {
+        continue;
+      }
+      if (size >= 500)
+      {
+        break;
+      }
+
+      size++;
+      fs << std::fixed << std::setprecision(2);
+      fs << std::to_string(pg.getMonoMass()) << "\t" << pg.getIntensity() << "\t" << pg.getRepCharge()
+                                                                                //  << "\t" << log10(pg.precursorSNR+1e-10) << "\t" << log10(pg.precursorTotalSNR+1e-10)
+                                                                                //  << "\t" << log10(pg.maxSNR + 1e-10) << "\t" << log10(pg.totalSNR + 1e-10)
+                                                                                << "\n";
+      fs << std::setprecision(-1);
+    }
+
+    fs << "END IONS\n\n";
+  }
+
+  bool DeconvolutedSpectrum::registerPrecursor(DeconvolutedSpectrum &precursorSpectrum)
+  {
+    //precursorSpectrum.updatePeakGroupMap();
+    for (auto &p: spec.getPrecursors())
+    {
+      for (auto &act :  p.getActivationMethods())
+      {
+        activationMethod = Precursor::NamesOfActivationMethodShort[act];
+        break;
+      }
+      precursorPeak = p;
+      precursorScanNumber = precursorSpectrum.scanNumber;
+      auto startMz = p.getIsolationWindowLowerOffset() > 100.0 ?
+                     p.getIsolationWindowLowerOffset() :
+                     -p.getIsolationWindowLowerOffset() + p.getMZ();
+      auto endMz = p.getIsolationWindowUpperOffset() > 100.0 ?
+                   p.getIsolationWindowUpperOffset() :
+                   p.getIsolationWindowUpperOffset() + p.getMZ();
+
+      double maxSumIntensity = 0.0;
+      for (auto &pg: precursorSpectrum)
```

Comment:
can this be const here? e.g. no sorting is required? because then the whole method could take a const DeconvolutedSpectrum&

Reply:
It caused problems in assigning pointers..